src/notebook/cells/widget.ts

```diff
@@ -378,23 +439,47 @@ class CodeCellWidget extends BaseCellWidget {
    * Handle `update_request` messages.
    */
   protected onUpdateRequest(message: Message): void {
-    this.toggleClass(COLLAPSED_CLASS, this._collapsedCursor.getValue());
-    // TODO: handle scrolled state.
-    this._output.trusted = this.trusted;
+    if (this._collapsedCursor) {
+      this.toggleClass(COLLAPSED_CLASS, this._collapsedCursor.getValue());
+    }
+    if (this._output) {
+      // TODO: handle scrolled state.
+      this._output.trusted = this.trusted;
+    }
     super.onUpdateRequest(message);
   }
 
   /**
+   * Handle the widget receiving a new model.
+   */
+  protected onModelChanged(msg: Message): void {
+    let model = this.model as ICodeCellModel;
+    let factory = this._factory;
+
+    if (this._output) {
+      this._output.dispose();
```

Comment:
This line will be replaced with a more efficient version where we simply replace the `.outputs` of the output area widget when it supports that. For the time being, it's being replaced wholesale.

cc: @blink1073